google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ConnectionImpl.java

```diff
@@ -995,33 +995,63 @@ GetQueryResultsResponse getQueryResultsFirstPage(JobId jobId) {
                 jobId.getLocation() == null && bigQueryOptions.getLocation() != null
                     ? bigQueryOptions.getLocation()
                     : jobId.getLocation());
-    try {
-      GetQueryResultsResponse results =
-          BigQueryRetryHelper.runWithRetries(
-              () ->
-                  bigQueryRpc.getQueryResultsWithRowLimit(
-                      completeJobId.getProject(),
-                      completeJobId.getJob(),
-                      completeJobId.getLocation(),
-                      connectionSettings.getMaxResultPerPage()),
-              bigQueryOptions.getRetrySettings(),
-              BigQueryBaseService.BIGQUERY_EXCEPTION_HANDLER,
-              bigQueryOptions.getClock(),
-              retryConfig);
 
-      if (results.getErrors() != null) {
-        List<BigQueryError> bigQueryErrors =
-            results.getErrors().stream()
-                .map(BigQueryError.FROM_PB_FUNCTION)
-                .collect(Collectors.toList());
-        // Throwing BigQueryException since there may be no JobId and we want to stay consistent
-        // with the case where there there is a HTTP error
-        throw new BigQueryException(bigQueryErrors);
+    // Implementing logic to poll the Job's status using getQueryResults as
+    // we do not get rows, rows count and schema unless the job is complete
+    // Ref: b/241134681
+    // This logic will wait for approx (poolingIntervalMs + 10 seconds which is the default timeout
+    // for getQueryResults) per iteration of the loop
+    long startTimeMs = System.currentTimeMillis();
+    long totalTimeOutMs = 18 * 60 * 60 * 1000; // 18 hours, which is the max timeout for the job
```

Comment:
I think it's reasonable to leave the totalTimeoutMs and assorted logic out of this PR.  You may in the future want to add a user configurable timeout, but per-rpc timeouts are sufficient.

BQ guarantees that jobs make forward progress (a job won't get stuck in pending forever).  This interval is so long you may as well just trust the system.

Reply:
sure. I have updated the logic and have added a `timeoutMs` param at the RPC layer which is currently hardcoded to 60 seconds. QQ, in the very corner case when the job runs for 18 hours and is still not-complete then will the backend throw and error? Otherwise it will be an infinite loop as `jobComplete` will never be `true`. 
Also, I have captured task to make `timeoutMs` user configurable here: https://github.com/googleapis/java-bigquery/issues/2240 (point#5)